File: java-sdk/lib/src/main/java/models/shared/Security.java
package .models.shared;

import .utils.SpeakeasyMetadata;

public class Security {
    @SpeakeasyMetadata("security:scheme=true,type=oauth2")
    public SchemeOAuth2Security oAuth2Security;
    public Security withOAuth2Security(SchemeOAuth2Security oAuth2Security) {
        this.oAuth2Security = oAuth2Security;
        return this;
    }
}

File: ts-sdk/src/sdk/models/shared/security.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class SchemeOAuth2Security extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, name=Authorization" })
  authorization: string;
}


export class Security extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oAuth2Security: SchemeOAuth2Security;
}
